### DISCUSSION TITLE
Mutation cache is not persisted

### DISCUSSION THREAD

**Discussion by nikl21** [TanStack/query#4343] [Q&A]
Hey I've been trying to implement react-query in my react native application.
So far so good and i've been able to implement the offline behaviour for queries successfully.
However I am not able to persist the mutations and it seems like the mutation cache is cleared everytime I close and reopen the app.
The offline mutations are not resumed if I close and reopen the app.
I am using PersistQueryClientProvider with async storage and also setting a mutationCache time of 5 minutes.
Is this the default behaviour or is there any way to resume the mutations once i kill the app and reconnect to the internet?

**Comment by TkDodo** [comment]
Impossible to say without seeing a reproduction. Have you compared your code against the offline example, because it works there:

https://tanstack.com/query/v4/docs/examples/react/offline
